xknx/config/schema.py

```diff
@@ -0,0 +1,606 @@
+                {
+                    vol.Required(CONF_ADDRESS): ensure_group_address,
+                    vol.Remove(CONF_STATE_ADDRESS): ensure_group_address,
+                    vol.Optional(CONF_STATE_UPDATE, default=False): False,
+                    vol.Required(CONF_SCENE_NUMBER): vol.All(
+                        vol.Coerce(int), vol.Range(min=1, max=64)
+                    ),
+                }
+            ),
+        }
+    )
+
+
+class NotificationSchema:
+    """Voluptuous schema for KNX notifications."""
+
+    SCHEMA = BaseDeviceSchema.SCHEMA.extend(
+        {
+            vol.Required(CONF_ADDRESS): RemoteValueSchema.SCHEMA.extend(
+                {
+                    vol.Required(CONF_ADDRESS): ensure_group_address,
+                }
+            ),
+        }
+    )
+
+
+class ExposeSchema:
+    """Voluptuous schema for KNX exposures."""
+
+    CONF_TYPE = "type"
+    CONF_ENTITY_ID = "entity_id"  # HA only
+    CONF_ATTRIBUTE = "attribute"
+    CONF_DEFAULT = "default"
+    CONF_NAME = "name"
+
+    SCHEMA = BaseDeviceSchema.SCHEMA.extend(
+        {
+            vol.Remove(CONF_NAME): False,
+            vol.Required(CONF_ADDRESS): ensure_group_address,
+            vol.Required(CONF_TYPE): sensor_value_type,
+            vol.Optional(CONF_DEFAULT): match_all,
+            vol.Optional(CONF_ATTRIBUTE): string,
+            vol.Optional(CONF_ENTITY_ID): valid_entity_id,
+        }
+    )
+
+
+class SensorSchema:
+    """Voluptuous schema for KNX sensors."""
+
+    CONF_TYPE = "type"
+    CONF_SENSOR = "sensor"
+
+    SCHEMA = BaseDeviceSchema.SCHEMA.extend(
+        {
+            vol.Required(CONF_SENSOR): RemoteValueSchema.SCHEMA.extend(
+                {
+                    vol.Required(CONF_STATE_ADDRESS): ensure_group_address,
+                    vol.Remove(CONF_ADDRESS): ensure_group_address,
+                    vol.Required(CONF_TYPE): sensor_value_type,
+                }
+            ),
+        }
+    )
+
+
+class XKNXSchema:
+    """Voloptuous schema for XKNX config."""
+
+    CONF_OWN_ADDRESS = "own_address"
+    CONF_RATE_LIMIT = "rate_limit"
+    CONF_VERSION = "version"
+    CONF_LOG_DIRECTORY = "log_directory"
+    CONF_FIRE_EVENT = "fire_event"
+    CONF_FIRE_EVENT_FILTER = "fire_event_filter"
+    CONF_MULTICAST_GROUP = "multicast_group"
+    CONF_MULTICAST_PORT = "multicast_port"
+
+    CONF_CONNECTION = "connection"
+    CONF_BINARY_SENSOR = "binary_sensor"
+    CONF_SWITCH = "switch"
+    CONF_LIGHT = "light"
+    CONF_FAN = "fan"
+    CONF_COVER = "cover"
+    CONF_CLIMATE = "climate"
+    CONF_WEATHER = "weather"
+    CONF_DATETIME = "datetime"
+    CONF_NOTIFICATION = "notification"
+    CONF_SCENE = "scene"
+    CONF_EXPOSE = "expose_sensor"
+    CONF_SENSOR = "sensor"
+
+    SCHEMA = vol.Schema(
+        {
+            vol.Optional(CONF_RATE_LIMIT, default=20): vol.All(
+                vol.Coerce(int), vol.Range(min=1, max=100)
```

Comment:
```suggestion
            vol.Optional(CONF_RATE_LIMIT, default=20): vol.All(
                vol.Coerce(int), vol.Range(min=0, max=100)
```
we can use 0 to disable rate limiter.